### PR TITLE
Filter out corrupted data

### PR DESCRIPTION
**Pull request by stkem**
I accidentally introduced some corrupted “allScores” data into the DB
with a very short lived previous PR (rolled back immediately and cause is fixed). This
will make _really_ sure it’s not surface to users.
@yingjieMiao @eishay

### CODE REVIEW

File: kifi-backend/modules/curator/app/com/keepit/curator/commanders/RecommendationRetrievalCommander.scala [hunk 1]
```diff
@@ -111,8 +111,19 @@ class RecommendationRetrievalCommander @Inject() (
...
+      reco.allScores.socialScore < 1000 &&
+        reco.allScores.popularityScore < 1000 &&
+        reco.allScores.overallInterestScore < 1000 &&
+        reco.allScores.recentInterestScore < 1000 &&
+        reco.allScores.recencyScore < 1000 &&
+        reco.allScores.priorScore < 1000 &&
+        reco.allScores.rekeepScore < 1000 &&
+        reco.allScores.discoveryScore < 1000
+    }
+
```
Comment (yingjieMiao) [inline kifi-backend/modules/curator/app/com/keepit/curator/commanders/RecommendationRetrievalCommander.scala:124]: hmm, your change was `x  => x * 10000 * 10000`. 
If all original subscores are less than `1e-5`, they are going to pass this guard. 
It's probably not critical.

Comment (yingjieMiao) [inline kifi-backend/modules/curator/app/com/keepit/curator/commanders/RecommendationRetrievalCommander.scala:124]: And looks like this method is really testing `IsGoodData`?

Comment (stkem) [inline kifi-backend/modules/curator/app/com/keepit/curator/commanders/RecommendationRetrievalCommander.scala:124]: Ha, yeah, it's not the best. I was moving very quickly :-)
It's a stop gap. I'll make it better.

Comment (stkem) [inline kifi-backend/modules/curator/app/com/keepit/curator/commanders/RecommendationRetrievalCommander.scala:124]: The name was mostly so I can write "filter badData" :-)

**Pull request merged**